### PR TITLE
[Feat] The `contains` API

### PR DESCRIPTION
**Pull request by rhdong**
- Please refer to https://en.cppreference.com/w/cpp/container/unordered_map/contains
- https://github.com/NVIDIA-Merlin/HierarchicalKV/issues/176

### CODE REVIEW

File: include/merlin/core_kernels/contains.cuh [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+          found_flag = true;
+        }
+      }
+      int found_vote = g.ballot(found_flag);
+      if (found_vote) {
+        sm_founds[key_idx_block] = 1;
+      }
+    }
+
+    /* Step4: write `found` */
```
Comment (jiashuy) [inline include/merlin/core_kernels/contains.cuh:199]: I think the step4 can be move into step 3. As the communication through shared memory is unnecessary.

Comment (rhdong) [reply include/merlin/core_kernels/contains.cuh]: Agree, I has merged it.

**Review by jiashuy — APPROVED** [review]
LGTM

**Pull request merged**